### PR TITLE
Refactor Windows debug code

### PR DESCRIPTION
**Pull request by GustavoLCR**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

* Fix some handle leaks
* Fix some memory leaks
* Re-use handles if possible
* Avoid allocations inside loops in `w32_desc_list()`
* Use debug plugin functions instead of reimplementing logic in `io_debug.c` and `io_w32dbg.c` to maintain state consistency
* Remove unneeded code
* Rename some functions

**Test plan**

Open rizin with `-d program`, do debugging tasks

### CODE REVIEW

File: librz/io/p/io_w32dbg.c [hunk 1]
```diff
@@ -237,20 +167,9 @@ static int __close(RzIODesc *fd) {
...
-			int pid = atoi(cmd + 3);
-			if (pid > 0 && pid != wrap->pi.dwThreadId && pid != wrap->pi.dwProcessId) {
-				wrap->pi.hThread = OpenThread(PROCESS_ALL_ACCESS, FALSE, pid);
-				if (!wrap->pi.hThread) {
-					eprintf("Cannot attach to %d\n", pid);
-				}
-			}
-		}
 		return rz_str_newf("%lu", wrap->pi.dwProcessId);
 	} else {
```
Comment (wargio) [inline librz/io/p/io_w32dbg.c:174]: is this command line coming from the rizin shell?

File: librz/debug/p/native/windows/windows_debug.c [hunk 2]
```diff
@@ -988,13 +979,12 @@ int w32_dbg_wait(RzDebug *dbg, int pid) {
...
 		case EXIT_PROCESS_DEBUG_EVENT:
 		case EXIT_THREAD_DEBUG_EVENT: {
-			PTHREAD_ITEM th = __find_thread(dbg, tid);
+			PTHREAD_ITEM th = find_thread(dbg, tid);
 			if (th) {
 				th->bFinished = TRUE;
-				th->hThread = INVALID_HANDLE_VALUE;
 				th->dwExitCode = de.u.ExitThread.dwExitCode;
 			} else {
-				__r_debug_thread_add(dbg, pid, tid, INVALID_HANDLE_VALUE, de.u.CreateThread.lpThreadLocalBase, de.u.CreateThread.lpStartAddress, TRUE);
```
Comment (XVilka) [inline librz/debug/p/native/windows/windows_debug.c:997]: Is this change intended? Why?

File: librz/io/p/io_w32dbg.c (hunk 1 above)
Comment (GustavoLCR) [reply librz/io/p/io_w32dbg.c:174]: its the `R!` command

File: librz/debug/p/native/windows/windows_debug.c (hunk 2 above)
Comment (GustavoLCR) [reply librz/debug/p/native/windows/windows_debug.c:997]: Yes, there should never be a situation that a thread exited without rizin having kept track of its creation. Before it happened to the initial thread, this pr fixes that

File: librz/debug/p/native/windows/windows_debug.c [hunk 3]
```diff
@@ -9,10 +9,11 @@
 
 const DWORD wait_time = 1000;
 static RzList *lib_list = NULL;
-static PLIB_ITEM last_lib = NULL;
 
-#define w32_PROCESS_ALL_ACCESS (STANDARD_RIGHTS_REQUIRED | SYNCHRONIZE | 0xFFF)
-#define w32_THREAD_ALL_ACCESS  w32_PROCESS_ALL_ACCESS
+#define w32_PROCESS_ALL_ACCESS  (STANDARD_RIGHTS_REQUIRED | SYNCHRONIZE | 0xFFF)
```
Comment (XVilka) [inline librz/debug/p/native/windows/windows_debug.c]: Everything else is fine. I just wonder if it's better to make all caps here: `W32_PROCESS_ALL_ACCESS` or something like that. Otherwise it looks weird.

**Review by XVilka — APPROVED** [review]

File: librz/debug/p/native/windows/windows_debug.c (hunk 3 above)
Comment (GustavoLCR) [reply librz/debug/p/native/windows/windows_debug.c]: @XVilka It wasn't used anywhere anymore, so I just removed it

**Review by wargio — APPROVED** [review]

**Comment by XVilka** [comment]
There is a new message in the tests:
```
[XX] C:\projects\rizin\test\db\archos\windows-x64\dbg_rebase var rebase
ANSICON=1 RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -d -Qc 'aa
afv @ entry0
doc
afv @ entry0
' bins/pe/hello_world/hello_world_od64bit.exe
-- stdout
--- expected
+++ actual
@@ -1,3 +1,4 @@
+(4604) Created process 3832 (start @ 00007FF6203A1278) (teb @ 000000AC34BD0000)
 var int64_t var_20h @ rsp+0x48
 var int64_t var_8h @ rsp+0x68
 var int64_t var_10h @ rsp+0x70
-- stderr
Spawned new process with pid 4604, tid = 3832
bin.baddr 0x7ff6203a0000
Using 0x7ff6203a0000
asm.bits 64
[ ] Analyze all flags starting with sym. and entry0 (aa)[[x] Analyze all flags starting with sym. and entry0 (aa)
[**]                 C:\projects\rizin\test\db\asm\8051    13540 OK       922 BR        1 XX       17 FX
[XX] C:\projects\rizin\test\db\archos\windows-x64\dbg_rebase ref rebase
ANSICON=1 RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -d -Qc 'aa
?v `axt main~entry0[1]`-`e bin.baddr`
doc
?v `axt main~entry0[1]`-`e bin.baddr`
' bins/pe/hello_world/hello_world_od64bit.exe
-- stdout
--- expected
+++ actual
@@ -1,2 +1,3 @@
+(3660) Created process 4788 (start @ 00007FF6203A1278) (teb @ 000000E0448C1000)
 0x1203
 0x1203
-- stderr
Spawned new process with pid 3660, tid = 4788
bin.baddr 0x7ff6203a0000
Using 0x7ff6203a0000
asm.bits 64
[ ] Analyze all flags starting with sym. and entry0 (aa)[[x] Analyze all flags starting with sym. and entry0 (aa)
[XX] C:\projects\rizin\test\db\archos\windows-x64\dbg_rebase flag rebase
ANSICON=1 RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -d -Qc 'fs test
f testflag @ main+10
?v `f~testflag[0]`-`e bin.baddr`
doc
?v `f~testflag[0]`-`e bin.baddr`
' bins/pe/hello_world/hello_world_od64bit.exe
-- stdout
--- expected
+++ actual
@@ -1,2 +1,3 @@
+(5712) Created process 1048 (start @ 00007FF6203A1278) (teb @ 0000007CC7985000)
 0x100a
 0x100a
-- stderr
Spawned new process with pid 5712, tid = 1048
bin.baddr 0x7ff6203a0000
Using 0x7ff6203a0000
asm.bits 64
[XX] C:\projects\rizin\test\db\archos\windows-x64\dbg_rebase bp rebase
ANSICON=1 RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -d -Qc 'db main
?v main-`db~main[0]`
doc
?v main-`db~main[0]`
' bins/pe/hello_world/hello_world_od64bit.exe
-- stdout
--- expected
+++ actual
@@ -1,2 +1,3 @@
+(2064) Created process 4836 (start @ 00007FF6203A1278) (teb @ 00000089F890F000)
 0x0
 0x0
-- stderr
Spawned new process with pid 2064, tid = 4836
bin.baddr 0x7ff6203a0000
Using 0x7ff6203a0000
asm.bits 64
[XX] C:\projects\rizin\test\db\archos\windows-x64\dbg_rebase function rebase
ANSICON=1 RZ_NOPLUGINS=1 rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -d -Qc '?v main-`e bin.baddr`
doc
?v main-`e bin.baddr`
' bins/pe/hello_world/hello_world_od64bit.exe
-- stdout
--- expected
+++ actual
@@ -1,2 +1,3 @@
+(1536) Created process 1968 (start @ 00007FF6203A1278) (teb @ 0000001B60D8D000)
 0x1000
 0x1000
-- stderr
Spawned new process with pid 1536, tid = 1968
bin.baddr 0x7ff6203a0000
Using 0x7ff6203a0000
asm.bits 64
[**] C:\projects\rizin\test\db\archos\windows-x64\dbg_rebase    13540 OK       922 BR        5 XX       17 FX
[**] C:\projects\
```
Not sure if it's just a debug message and can be silenced, or tests updated.

**Comment by wargio** [comment]
set it via RZ_LOG_INFO ?
